submissions/scripts/export_lims_metadata.py

```diff
@@ -0,0 +1,190 @@
+    else:
+        print(f'Error:\n HTTP status {response.status_code}\n {response.content}')
+        response.raise_for_status()
+
+
+def get_lims_sequencing_metadata(barcodes: list):
+    body = json.dumps([{"anyBarcode": b} for b in barcodes if not str(b).endswith("_exp")])
+
+    response = lims_api_request(body, '/api/v1/sfs-specimens/sequencing-submission-metadata')
+
+    if response.content is not None:
+        df = pd.json_normalize(json.loads(response.content), max_level=3)
+
+        column_map = {
+            'specimen.ids.sampleIdentifierForDohReporting': 'sfs_identifier_for_doh_reporting',
+            'specimen.ids.sampleId': 'sfs_sample_identifier',
+            'specimen.ids.sampleBarcode': 'sfs_sample_barcode',
+            'specimen.ids.collectionBarcode': 'sfs_collection_barcode',
+            'specimen.origin.collection.collectionDate': 'collection_date',
+            'specimen.origin.collection.swabType': 'swab_type',
+            'specimen.origin.source': 'source',
+            'subject.pumaGeoId':'puma',
+            'subject.county':'county',
+            'subject.state': 'state',
+            'sequencing.sentinelSurveillance': 'baseline_surveillance',
+            'error.message': 'error_message',
+        }
+        df.rename(columns=column_map, inplace=True)
+
+        df['baseline_surveillance'] = df['baseline_surveillance'].map({True:'t', False:'f'})
+
+        invalid_addresses = df[df['error_message']== 'Address could not be validated.']
+        if not invalid_addresses.empty:
+            raise Exception(f"Error: invalid addresses: \n {invalid_addresses[['sfs_sample_identifier','sfs_sample_barcode']].to_json(orient='records')}")
+
+        return df
+    else:
+        raise Exception(f'Error:\n Empty response for: {body}')
+
+
+def get_lims_additional_sequencing_metadata(barcodes: list):
+    body = json.dumps([{"anyBarcode": b} for b in barcodes if not str(b).endswith("_exp")])
+    response = lims_api_request(body, '/api/v1/sfs-specimens/sequencing-submission-metadata-for-or-ha')
+
+    if response.content is not None:
+        df = pd.json_normalize(json.loads(response.content), max_level=3)
+
+        column_map = {
+            'subject.participantId': 'participant_id',
+            'subject.firstName': 'first_name',
+            'subject.lastName': 'last_name',
+            'subject.birthdate': 'birthdate',
+            'subject.sex': 'sex',
+            'subject.address.street': 'street',
+            'subject.address.city': 'city',
+            'subject.address.state': 'state',
+            'subject.address.zipCode': 'zip',
+            'subject.phone': 'phone',
+            'specimen.ids.sampleId': 'sfs_sample_identifier',
+            'specimen.ids.sampleBarcode': 'sfs_sample_barcode',
+            'specimen.ids.collectionBarcode': 'sfs_collection_barcode',
+            'specimenCollectionSite': 'specimen_collection_site'
+        }
+        df.rename(columns=column_map, inplace=True)
```

Comment:
Might want to add a check here that the df is non-empty before trying to rename, since if you try to rename an empty dataframe the column names won't show up and you'll get an index error down the line. This was happening to me with the 8-10 batch. Something like the code below is what I was using to get it working:
```
        if df.empty:
            df = pd.DataFrame(columns=column_map.values())
        else:
            df.rename(columns=column_map, inplace=True)
```

Reply:
Thanks for catching that! Looks like that should only occur when there are no Oregon samples, in which case we can skip the call to `add_lims_additional_metadata` entirely.  I'll make that change.